test/kubernetes/e2e/features/services/tcproute/suite.go

```diff
@@ -31,95 +37,192 @@ func NewTestingSuite(ctx context.Context, testInst *e2e.TestInstallation) suite.
-	s.testInstallation.Assertions.EventuallyTCPRouteCondition(s.ctx, "tcp-app-2", "default", v1.RouteConditionAccepted, metav1.ConditionTrue, timeout)
-	s.testInstallation.Assertions.EventuallyTCPRouteCondition(s.ctx, "tcp-app-2", "default", v1.RouteConditionResolvedRefs, metav1.ConditionTrue, timeout)
-	s.testInstallation.Assertions.EventuallyGatewayCondition(s.ctx, "tcp-gateway", "default", v1.GatewayConditionProgrammed, metav1.ConditionTrue, timeout)
-	s.testInstallation.Assertions.EventuallyGatewayListenerAttachedRoutes(s.ctx, "tcp-gateway", "default", v1.SectionName("foo"), int32(1), timeout)
-	s.testInstallation.Assertions.EventuallyGatewayListenerAttachedRoutes(s.ctx, "tcp-gateway", "default", v1.SectionName("bar"), int32(1), timeout)
-
-	s.testInstallation.Assertions.AssertEventualCurlResponse(
-		s.ctx,
-		defaults.CurlPodExecOpt,
-		[]curl.Option{
-			curl.WithHost(kubeutils.ServiceFQDN(proxyService.ObjectMeta)),
-			curl.WithPort(8088),
+func (s *testingSuite) TestConfigureTCPRouteBackingDestinations() {
+	testCases := []tcpRouteTestCase{
+		{
+			name:             "SingleServiceTCPRoute",
+			nsManifest:       singleSvcNsManifest,
+			gtwName:          singleSvcGatewayName,
+			gtwNs:            singleSvcNsName,
+			gtwManifest:      singleSvcGatewayAndClientManifest,
+			svcManifest:      singleSvcBackendManifest,
+			tcpRouteManifest: singleSvcTcpRouteManifest,
+			proxyService:     singleSvcProxyService,
+			proxyDeployment:  singleSvcProxyDeployment,
+			expectedResponses: []*matchers.HttpResponse{
+				expectedSingleSvcResp,
+			},
+			ports: []int{8087},
+			listenerNames: []v1.SectionName{
+				v1.SectionName(singleSvcListenerName8087),
+			},
+			expectedRouteCounts: []int32{1},
+			tcpRouteNames:       []string{singleSvcTCPRouteName},
 		},
-		expectedTcpFooSvcResp)
-	s.testInstallation.Assertions.AssertEventualCurlResponse(
-		s.ctx,
-		defaults.CurlPodExecOpt,
-		[]curl.Option{
-			curl.WithHost(kubeutils.ServiceFQDN(proxyService.ObjectMeta)),
-			curl.WithPort(8089),
+		{
+			name:             "MultiServicesTCPRoute",
+			nsManifest:       multiSvcNsManifest,
+			gtwName:          multiSvcGatewayName,
+			gtwNs:            multiSvcNsName,
+			gtwManifest:      multiSvcGatewayAndClientManifest,
+			svcManifest:      multiSvcBackendManifest,
+			tcpRouteManifest: multiSvcTcpRouteManifest,
+			proxyService:     multiProxyService,
+			proxyDeployment:  multiProxyDeployment,
+			expectedResponses: []*matchers.HttpResponse{
+				expectedMultiSvc1Resp,
+				expectedMultiSvc2Resp,
+			},
+			ports: []int{8088, 8089},
+			listenerNames: []v1.SectionName{
+				v1.SectionName(multiSvcListenerName8088),
+				v1.SectionName(multiSvcListenerName8089),
+			},
+			expectedRouteCounts: []int32{1, 1},
+			tcpRouteNames:       []string{multiSvcTCPRouteName1, multiSvcTCPRouteName2},
 		},
-		expectedTcpBarSvcResp)
+	}
+
+	for _, tc := range testCases {
+		tc := tc // capture range variable
+		s.Run(tc.name, func() {
+			s.T().Parallel()
```

Comment:
I like the idea here. Though I wonder if we gain more from making suites run serially, and then run a set of suites in parallel against an installation. wdyt?